packages/doenetml-worker-rust/lib-js-wasm-binding/src/eval-math.ts

```diff
@@ -18,9 +24,128 @@ export function evalWithMathExpressionsInScope(source: string) {
                 "Running with MathExpressions in scope",
                 MathExpressions,
                 window,
+                serializedComponentsReviver,
             );
         }
-        return JSON.stringify(eval(source));
+
+        return JSON.stringify(eval(source), serializedComponentsReplacer);
     }
     return _eval(me, globalThis);
 }
+
+/**
+ * Parse the string `text` into math using the `math-expressions` text parser.
+ * Return the `JSON.stringify`ed results.
+ */
+export function parseTextIntoMath(
+    text: string,
+    splitSymbols: boolean,
+    functionSymbols: string[],
+    parseScientificNotation: boolean,
+) {
+    let fromText = textToMathFactory({
+        splitSymbols: splitSymbols,
+        parseScientificNotation: parseScientificNotation,
+        functionSymbols,
+    });
+    let expression;
+    try {
+        expression = fromText(text);
+    } catch (e) {
+        expression = me.fromAst("\uFF3F"); // long underscore
+        console.warn("Invalid value for a math of text format: " + text);
+    }
+
+    return JSON.stringify(expression, serializedComponentsReplacer);
+}
+
+/**
+ * Parse the string `latex` into math using the `math-expressions` latex parser.
+ * Return the `JSON.stringify`ed results.
+ */
```

Comment:
Can you explain what the arguments are? You can make a list with `@argName` syntax. E.g.


```typescript
/**
 * @latex - the source string
 * @splitSymbols - ???
 ...
 */
```

Reply:
done